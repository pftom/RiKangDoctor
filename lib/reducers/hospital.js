Object.defineProperty(exports,"__esModule",{value:true});var _immutable=require('immutable');var _immutable2=_interopRequireDefault(_immutable);


var _constants=require('../constants/');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}















var initialHospitalValue=_immutable2.default.Map({
hospitals:null,
hospital:null,
hospitalDoctors:null,
isLoadingData:false,
loadingError:false,
loadingSuccess:false});


var hospitalValue=function hospitalValue(){var state=arguments.length>0&&arguments[0]!==undefined?arguments[0]:initialHospitalValue;var action=arguments[1];
switch(action.type){
case _constants.GET_HOSPITALS:
case _constants.GET_SINGLE_HOSPITAL:
case _constants.GET_SINGLE_HOSPITAL_DOCTORS:

return state.merge({
isLoadingData:true,
loadingError:false,
loadingSuccess:false});


case _constants.GET_HOSPITALS_SUCCESS:var

hospitals=action.hospitals;
console.log('hospitals',hospitals);
return state.merge({
isLoadingData:false,
loadingSuccess:true,
hospitals:hospitals});


case _constants.GET_SINGLE_HOSPITAL_SUCCESS:var

hospital=action.hospital;
return state.merge({
isLoadingData:false,
loadingSuccess:true,
hospital:hospital});


case _constants.GET_SINGLE_HOSPITAL_DOCTORS_SUCCESS:var

hospitalDoctors=action.hospitalDoctors;
return state.merge({
isLoadingData:false,
loadingSuccess:true,
hospitalDoctors:hospitalDoctors});



case _constants.GET_HOSPITALS_ERROR:
case _constants.GET_SINGLE_HOSPITAL_ERROR:
case _constants.GET_SINGLE_HOSPITAL_DOCTORS_ERROR:

return state.merge({
isLoadingData:false,
loadingError:true});


default:
return state;}

};exports.default=

hospitalValue;