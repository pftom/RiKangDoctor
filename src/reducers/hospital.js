import Immutable from 'immutable';

//import single action constants
import {
  GET_HOSPITALS,
  GET_HOSPITALS_SUCCESS,
  GET_HOSPITALS_ERROR,

  GET_SINGLE_HOSPITAL,
  GET_SINGLE_HOSPITAL_SUCCESS,
  GET_SINGLE_HOSPITAL_ERROR,

  GET_SINGLE_HOSPITAL_DOCTORS,
  GET_SINGLE_HOSPITAL_DOCTORS_SUCCESS,
  GET_SINGLE_HOSPITAL_DOCTORS_ERROR,
} from '../constants/'


//home single reducers
const initialHospitalValue = Immutable.Map({
  hospitals: null,
  hospital: null,
  hospitalDoctors: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: false,
});

const hospitalValue = (state = initialHospitalValue, action) => {
    switch (action.type) {
      case GET_HOSPITALS:
      case GET_SINGLE_HOSPITAL:
      case GET_SINGLE_HOSPITAL_DOCTORS:

        return state.merge({
          isLoadingData: true,
          loadingError: false,
          loadingSuccess: false,
        });
      
      case GET_HOSPITALS_SUCCESS:

        const { hospitals } = action;
        return state.merge({
          isLoadingData: false,
          loadingSuccess: true,
          hospitals,
        });
      
      case GET_SINGLE_HOSPITAL_SUCCESS:

      const { hospital } = action;
      return state.merge({
        isLoadingData: false,
        loadingSuccess: true,
        hospital,
      });

      case GET_SINGLE_HOSPITAL_DOCTORS_SUCCESS:

      const { hospitalDoctors } = action;
      return state.merge({
        isLoadingData: false,
        loadingSuccess: true,
        hospitalDoctors,
      });
        

      case GET_HOSPITALS_ERROR:
      case GET_SINGLE_HOSPITAL_ERROR:
      case GET_SINGLE_HOSPITAL_DOCTORS_ERROR:

        return state.merge({
          isLoadingData: false,
          loadingError: true,
        });

      default:
        return state;
  }
};

export default hospitalValue;