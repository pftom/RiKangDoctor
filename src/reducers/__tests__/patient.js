import Immutable from 'immutable';

import hospital from '../hospital';

//import single action constants
import {
  GET_HOSPITALS,
  GET_HOSPITALS_SUCCESS,
  GET_HOSPITALS_ERROR,

  GET_SINGLE_HOSPITAL,
  GET_SINGLE_HOSPITAL_SUCCESS,
  GET_SINGLE_HOSPITAL_ERROR,

  GET_SINGLE_HOSPITAL_DOCTORS,
  GET_SINGLE_HOSPITAL_DOCTORS_SUCCESS,
  GET_SINGLE_HOSPITAL_DOCTORS_ERROR,
} from '../../constants/';


//get hospitals
const getHospitalsAction = {
  type: GET_HOSPITALS,
};

const getHospitalsSuccessAction = {
  type: GET_HOSPITALS_SUCCESS,
  hospitals: {},
};

const getHospitalsErrorAction = {
  type: GET_HOSPITALS_ERROR,
};

//get single hospital
const getSingleHospitalAction = {
  type: GET_SINGLE_HOSPITAL,
};

const getSingleHospitalSuccessAction = {
  type: GET_SINGLE_HOSPITAL_SUCCESS,
  hospital: {},
};

const getSingleHospitalErrorAction = {
  type: GET_SINGLE_HOSPITAL_ERROR,
};

//get posts
const getSingleHospitalDoctorsAction = {
  type: GET_SINGLE_HOSPITAL_DOCTORS,
};

const getSingleHospitalDoctorsSuccessAction = {
  type: GET_SINGLE_HOSPITAL_DOCTORS_SUCCESS,
  hospitalDoctors: {},
};

const getSingleHospitalDoctorsErrorAction = {
  type: GET_SINGLE_HOSPITAL_DOCTORS_ERROR,
};


//home single reducers
const initialHospitalValue = Immutable.Map({
  hospitals: null,
  hospital: null,
  hospitalDoctors: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: false,
});

test('the hospital reducer work as well', () => {
  //test hospitals
  expect(hospital(initialHospitalValue, getHospitalsAction)).toMatchSnapshot();
  expect(hospital(initialHospitalValue.merge({ isLoadingData: true }), getHospitalsSuccessAction)).toMatchSnapshot();
  expect(hospital(initialHospitalValue.merge({ isLoadingData: true }), getHospitalsErrorAction)).toMatchSnapshot();

  //test hospital
  expect(hospital(initialHospitalValue, getSingleHospitalAction)).toMatchSnapshot();
  expect(hospital(initialHospitalValue.merge({ isLoadingData: true }), getSingleHospitalSuccessAction)).toMatchSnapshot();
  expect(hospital(initialHospitalValue.merge({ isLoadingData: true }), getSingleHospitalErrorAction)).toMatchSnapshot();

  //test hospital doctors
  expect(hospital(initialHospitalValue, getSingleHospitalDoctorsAction)).toMatchSnapshot();
  expect(hospital(initialHospitalValue.merge({ isLoadingData: true }), getSingleHospitalDoctorsSuccessAction)).toMatchSnapshot();
  expect(hospital(initialHospitalValue.merge({ isLoadingData: true }), getSingleHospitalDoctorsErrorAction)).toMatchSnapshot();

});