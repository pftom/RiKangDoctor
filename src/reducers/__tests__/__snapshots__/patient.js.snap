// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the patient reducer work as well 1`] = `
Immutable.Map {
  doctorFav: Immutable.List [
  ],
  postFav: Immutable.List [
  ],
  questionStar: Immutable.List [
  ],
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the patient reducer work as well 2`] = `
Immutable.Map {
  doctorFav: Immutable.List [
    Object {},
  ],
  postFav: Immutable.List [
  ],
  questionStar: Immutable.List [
  ],
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the patient reducer work as well 3`] = `
Immutable.Map {
  doctorFav: Immutable.List [
  ],
  postFav: Immutable.List [
  ],
  questionStar: Immutable.List [
  ],
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;

exports[`the patient reducer work as well 4`] = `
Immutable.Map {
  doctorFav: Immutable.List [
  ],
  postFav: Immutable.List [
  ],
  questionStar: Immutable.List [
  ],
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the patient reducer work as well 5`] = `
Immutable.Map {
  doctorFav: Immutable.List [
  ],
  postFav: Immutable.List [
  ],
  questionStar: Immutable.List [
    Object {},
  ],
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the patient reducer work as well 6`] = `
Immutable.Map {
  doctorFav: Immutable.List [
  ],
  postFav: Immutable.List [
  ],
  questionStar: Immutable.List [
  ],
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;
